### PR TITLE
Newsletter: Remove color option from onboarding

### PR DESCRIPTION
**Pull request by edanzer**
## Proposed Changes

- Remove the color option from the Newsletter Setup screen during onboarding.
- Remove all associated code. I decided to be thorough and remove any associated, now-unused code. This includes the AccentColorControl component and two custom hook files. We could do a simpler PR and just remove the control itself, and leave other code in place in case we need it again in the future. 
- Updated launchpad tests to reflect changes.

color control

## Testing Instructions

We're testing two things: first, the color control does not show and, second, there are no regressive breakages from removing so much code. 

1) Check out this branch and start a newsletter site at http://calypso.localhost:3000/setup/newsletter/intro
2) Confirm the color control no longer shows on the Setup screen, and that the screen otherwise works as expected.
   - The rest of testing will be simpler if you do not choose the paid newsletter option.
3) Go to Launchpad. Confirm that the launchpad screen, task list, and web preview all load as expected.
   - Note we still had to supply a default color for site creation, which is blue. So you'll see blue colors on the web preview.
4) Click the Personalize your newsletter task. This will take you to the 'Post Setup' screen. Confirm that the color control does not show and that the screen otherwise works as expected. 
5) Go back to launchpad, click the Start Writing task and publish a post. This should launch the site. Confirm that you can access My Home (you are not redirected to Launchpad). 
6) Confirm Launchpad tests still pass by running `yarn run test-client client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/` (I updated tests because we removed some params from a key method).

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by simison** [comment]
@Copons @mmtr  :wave: are the global styles bits in the sidebar used only for Newsletters, or do other flows use them as well?

Thinking e.g. design flows if users pick style variations?

**Comment by lezama** [comment]
for the Newsletter case this tests well without errors, I will let the Calypso experts give the 🟢 light

**Comment by mmtr** [comment]
> are the global styles bits in the sidebar used only for Newsletters, or do other flows use them as well?

Good catch! They are used in other flows, so these changes will affect them if the user picks a custom style variation and doesn't upgrade:

Flow | Before (Production) | After (This PR)
--- | --- | ---
`/start` | Screenshot 2023-07-13 at 11 19 33 | Screenshot 2023-07-13 at 11 20 08

**Comment by edanzer** [comment]
@simison - Good eyes and question on the launchpad global styles changes. And @mmtr - Thanks for taking a careful look and catching the imp. 

We obviously don't want to affect other flows. I've reverted changes to launchpad sidebar and task-helpers and the associated tests.

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx [hunk 1]
```diff
@@ -147,7 +145,10 @@ const SiteCreationStep: Step = function SiteCreationStep( { navigation, flow, da
 			theme,
 			siteVisibility,
 			urlData?.meta.title ?? selectedSiteTitle,
-			siteAccentColor,
+			// We removed the color option during newsletter onboarding.
+			// But backend still expects/needs a value, so supplying the default.
+			// Ideally should remove this and update code downstream to handle this.
+			'#113AF5',
```
Comment (simison) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/index.tsx:154]: Let's follow up with backend to not send this and clean up afterwards. AFAIK other flows aren't using this.  👍

**Comment by simison** [comment]
Added string freeze to get that edited string translated; it might get marked "fuzzy" translated which is fine to get this shipped I think.

**Review by simison — APPROVED** [review]
Tested `/setup/newsletter` and works well. Also checked post-launchpad setup screen.

Screenshot 2023-07-13 at 18 40 38

Tested also that we have styles in the site-editor, and that default style was selected after onboarding (the blue one):

image

Screenshot 2023-07-13 at 18 41 52

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8549665

Thank you @edanzer for including a screenshot in the description! This is really helpful for our translators.

**Comment by mmtr** [comment]
> We obviously don't want to affect other flows. I've reverted changes to launchpad sidebar and task-helpers and the associated tests.

Thank you! It works as expected now

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8549665).

**Pull request merged**